### ISSUE TITLE
`opam lock .` returns "Not all dependencies are satisfied, won't include them"

### ISSUE THREAD

**Issue by 3Rafal** [ocaml/opam#5586]
Repository: https://github.com/mirage/irmin
Commit: 8da4d16e7cc8beddfc8a824feca325426bae08a9

after installing all the dependencies without error, opam lock reports that not all dependencies are satisfied:
```
rafal@memepad:~/Projects/Tarides/irmin$ opam install . --with-test
[irmin-bench.3.7.2] synchronised (no changes)
[irmin.3.7.2] synchronised (no changes)
[irmin-chunk.3.7.2] synchronised (no changes)
[irmin-cli.3.7.2] synchronised (no changes)
[irmin-containers.3.7.2] synchronised (no changes)
[irmin-fs.3.7.2] synchronised (no changes)
[irmin-graphql.3.7.2] synchronised (no changes)
[irmin-git.3.7.2] synchronised (no changes)
[irmin-http.3.7.2] synchronised (no changes)
[irmin-mirage-git.3.7.2] synchronised (no changes)
[irmin-mirage.3.7.2] synchronised (no changes)
[irmin-mirage-graphql.3.7.2] synchronised (no changes)
[irmin-pack-tools.3.7.2] synchronised (no changes)
[irmin-pack.3.7.2] synchronised (no changes)
[irmin-test.3.7.2] synchronised (no changes)
[irmin-tezos.3.7.2] synchronised (no changes)
[ppx_irmin.3.7.2] synchronised (no changes)
[libirmin.3.7.2] synchronised (no changes)
[NOTE] Package irmin-bench is already installed (current version is 3.7.2).
[NOTE] Package irmin-chunk is already installed (current version is 3.7.2).
[NOTE] Package irmin-cli is already installed (current version is 3.7.2).
[NOTE] Package irmin-containers is already installed (current version is 3.7.2).
[NOTE] Package irmin-fs is already installed (current version is 3.7.2).
[NOTE] Package irmin-git is already installed (current version is 3.7.2).
[NOTE] Package irmin-graphql is already installed (current version is 3.7.2).
[NOTE] Package irmin-http is already installed (current version is 3.7.2).
[NOTE] Package irmin-mirage-git is already installed (current version is 3.7.2).
[NOTE] Package irmin-mirage-graphql is already installed (current version is 3.7.2).
[NOTE] Package irmin-mirage is already installed (current version is 3.7.2).
[NOTE] Package irmin-pack-tools is already installed (current version is 3.7.2).
[NOTE] Package irmin-pack is already installed (current version is 3.7.2).
[NOTE] Package irmin-test is already installed (current version is 3.7.2).
[NOTE] Package irmin-tezos is already installed (current version is 3.7.2).
[NOTE] Package irmin is already installed (current version is 3.7.2).
[NOTE] Package libirmin is already installed (current version is 3.7.2).
[NOTE] Package ppx_irmin is already installed (current version is 3.7.2).
rafal@memepad:~/Projects/Tarides/irmin$ opam lock .
Not all dependencies are satisfied, won't include them
Not all dependencies are satisfied, won't include them
Not all dependencies are satisfied, won't include them
Generated lock files for:
  - ppx_irmin.3.7.2: /home/rafal/Projects/Tarides/irmin/ppx_irmin.opam.locked
  - libirmin.3.7.2: /home/rafal/Projects/Tarides/irmin/libirmin.opam.locked
  - irmin-tezos.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-tezos.opam.locked
  - irmin-test.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-test.opam.locked
  - irmin-pack-tools.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-pack-tools.opam.locked
  - irmin-pack.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-pack.opam.locked
  - irmin-mirage-graphql.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-mirage-graphql.opam.locked
  - irmin-mirage-git.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-mirage-git.opam.locked
  - irmin-mirage.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-mirage.opam.locked
  - irmin-http.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-http.opam.locked
  - irmin-graphql.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-graphql.opam.locked
  - irmin-git.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-git.opam.locked
  - irmin-fs.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-fs.opam.locked
  - irmin-containers.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-containers.opam.locked
  - irmin-cli.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-cli.opam.locked
  - irmin-chunk.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-chunk.opam.locked
  - irmin-bench.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin-bench.opam.locked
  - irmin.3.7.2: /home/rafal/Projects/Tarides/irmin/irmin.opam.locked
```

**Comment by rjbou** [comment]
It seems that is fixed by #4993. Can you check?

**Comment by kit-ty-kate** [comment]
After trying to reproduce it i can confirm it's been fixed in opam 2.2.0.
The only output i'm getting now is the expected:
```
Not all dependencies of irmin.3.8.0 are satisfied, not including these: odoc
Generated lock files for:
  - [...]
```
